code/game/machinery/computer/warrant.dm

```diff
@@ -0,0 +1,220 @@
+				if("*Arrest*")
+					background = "background-color:#990000;"
+					notice = "<br>**REPORT TO THE BRIG**"
+				if("Incarcerated")
+					background = "background-color:#CD6500;"
+				if("Paroled")
+					background = "background-color:#CD6500;"
+				if("Discharged")
+					background = "background-color:#006699;"
+				if("None")
+					background = "background-color:#4F7529;"
+				if("")
+					background = "''" //"'background-color:#FFFFFF;'"
+			dat += "<font size='4'><b>Warrant Data</b></font>"
+			dat += {"<table>
+			<tr><td>Name:</td><td>&nbsp;[current.fields["name"]]&nbsp;</td></tr>
+			<tr><td>ID:</td><td>&nbsp;[current.fields["id"]]&nbsp;</td></tr>
+			</table>"}
+			dat += {"<br>Criminal Status:<br>
+			<div style='[background] padding: 3px; text-align: center;'>
+			<strong>[current.fields["criminal"]][notice]</strong>
+			</div>"}
+
+			dat += "<br><br>Citations:"
+
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Fine</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			<th>Amount Due</th>
+			<th>Make Payment</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["citation"])
+				var/owed = c.fine - c.paid
+				dat += "<tr><td>[c.crimeName]</td>"
+				dat += "<td>$[c.fine]</td>"
+				dat += "<td>[c.author]</td>"
+				dat += "<td>[c.time]</td>"
+				if(owed > 0)
+					dat += "<td>$[owed]</td>"
+					dat += "<td><A href='?src=[REF(src)];choice=Pay;field=citation_pay;cdataid=[c.dataId]'>\[Pay\]</A></td>"
+				else
+					dat += "<td colspan='2'>All Paid Off</td>"
+				dat += "</tr>"
+			dat += "</table>"
+
+			dat += "<br>Minor Crimes:"
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Details</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["mi_crim"])
+				dat += "<tr><td>[c.crimeName]</td>"
+				dat += "<td>[c.crimeDetails]</td>"
+				dat += "<td>[c.author]</td>"
+				dat += "<td>[c.time]</td>"
+				dat += "</tr>"
+			dat += "</table>"
+
+			dat += "<br>Major Crimes:"
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Details</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["ma_crim"])
+				dat += "<tr><td>[c.crimeName]</td>"
+				dat += "<td>[c.crimeDetails]</td>"
+				dat += "<td>[c.author]</td>"
+				dat += "<td>[c.time]</td>"
+				dat += "</tr>"
+			dat += "</table>"
+		else
+			dat += {"<span>** No security record found for this ID **</span>"}
+
+	dat = dat.Join()
+	var/datum/browser/popup = new(user, "warrant", "Security Warrant Console", 600, 400)
+	popup.set_content(dat)
+	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+
+/obj/machinery/computer/warrant/Topic(href, href_list)
+	if(..())
+		return
+
+	switch(href_list["choice"])
+		if("Login")
```

Comment:
This is just a suggestion but if this is meant to be public console, autoscan and just displaying their own to users will allow few people to use it at once, you don't really need to persist the record here.

Reply:
I did this because its withdrawing funds directly from the user's bank account on their ID. I'd like it if they could just hit the pay button with a holochip though. Actually that makes a lot more sense.

Reply:
You were right, the new way is much better.